lib/logstash/codecs/netflow.rb

```diff
@@ -83,161 +85,168 @@ def decode(payload, &block)
     end
 
     flowset.records.each do |record|
+      # for better style we could have created a single array event return for decode_netflow5
+      # and yield in a consistent way but this will perform a bit better for netflow 5
+
+      # I do not understand why we need to check flowset.version here since we already made that
+      # decision from the header. I will keep that logic the same here to avoid protocol
+      # parsing regression if this is indeed required for some reason.
+
       if flowset.version == 5
-        event = LogStash::Event.new
-
-        # FIXME Probably not doing this right WRT JRuby?
-        #
-        # The flowset header gives us the UTC epoch seconds along with
-        # residual nanoseconds so we can set @timestamp to that easily
-        event.timestamp = LogStash::Timestamp.at(flowset.unix_sec, flowset.unix_nsec / 1000)
-        event[@target] = {}
-
-        # Copy some of the pertinent fields in the header to the event
-        ['version', 'flow_seq_num', 'engine_type', 'engine_id', 'sampling_algorithm', 'sampling_interval', 'flow_records'].each do |f|
-          event[@target][f] = flowset[f]
-        end
+        yield(decode_netflow5(flowset, record))
+      elsif flowset.version == 9
+        decode_netflow9(flowset, record).each{|event| yield(event)}
```

Comment:
well, it might always be a single event, I don't know, but per the decode logic, it could generate more events since the event creation is in the `records.each` loop https://github.com/colinsurprenant/logstash-codec-netflow/blob/fix/json/lib/logstash/codecs/netflow.rb#L204-L231

anything I am missing?